### PR TITLE
feat(oauth): add ability to limit user IDs for Discord authentication

### PR DESCRIPTION
**Pull request by COMPLEXWASTAKEN**
hello. I have added a discord oauth id restriction to where you can set a set of ids like <id>,<id> and if there id is not there it will not allow them to register with discord oauth

lmk if you want anything changed or find bugs

### CODE REVIEW

File: backend/src/oauth/provider/discord.provider.ts [hunk 1]
```diff
@@ -81,7 +81,11 @@ export class DiscordProvider implements OAuthProvider<DiscordToken> {
     if (guild) {
       await this.checkLimitedGuild(token, guild);
     }
-
+    const limitedusers = this.config.get("oauth.discord-limitedUsers");
```
Comment (stonith404) [inline backend/src/oauth/provider/discord.provider.ts:84]: Sorry for being so picky but could you please rename the variable to `limitedUsers`?
```suggestion
    const limitedUsers = this.config.get("oauth.discord-limitedUsers");
```

**Review by stonith404 — CHANGES_REQUESTED** [review]
Thanks for your contribution :) 

The code looks good to me but could you please remove all translations and only keep the changes for the `en-US.ts` file. The strings will get translated by contributors on Crowdin.

**Comment by COMPLEXWASTAKEN** [comment]
oops i didnt mean to push the other to this pull request

**Comment by stonith404** [comment]
No problem, but could you please remove these changes from this PR? After that it should be ready to merge :)

Regarding the email restriction, I think this would need a bit more discussion as I've mentioned, the feature doesn't really make sense for email and password authentication.

**Comment by COMPLEXWASTAKEN** [comment]
How do I remove them I have never did a pull request on GitHub lol

**Comment by stonith404** [comment]
You have to do this in the Git CLI locally with the command `git revert <commit-that-you-want-to-revert>`. In this case it would be `git revert 91fb162558cffd5f8685f1eb606729874ff640db`. After that you can normally run `git push` and the commit should be reverted :)

**Comment by COMPLEXWASTAKEN** [comment]
done

**Review by stonith404 — APPROVED** [review]

**Pull request merged**